nixos/modules/services/networking/gns3-server.nix

```diff
@@ -0,0 +1,267 @@
+        package = lib.mkPackageOptionMD pkgs "vpcs" { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ]
+      ++ lib.optional cfg.dynamips.enable cfg.dynamips.package
+      ++ lib.optional cfg.ubridge.enable cfg.ubridge.package
+      ++ lib.optional cfg.vpcs.enable cfg.vpcs.package;
+
+    users.groups.ubridge = lib.mkIf cfg.ubridge.enable { };
+
+    security.wrappers.ubridge = lib.mkIf cfg.ubridge.enable {
+      capabilities = "cap_net_raw,cap_net_admin=eip";
+      group = "ubridge";
+      owner = "root";
+      permissions = "u=rwx,g=rx,o=r";
+      source = "${cfg.ubridge.package}/bin/ubridge";
+    };
+
+    users.users.gns3 = {
+      name = "gns3";
+      group = "gns3";
+      description = "GNS3 user";
+      isSystemUser = true;
+      home = "/var/lib/gns3";
+      extraGroups = lib.optional flags.enableDocker "docker"
+        ++ lib.optional flags.enableLibvirtd "libvirtd"
+        ++ lib.optional cfg.ubridge.enable "ubridge";
+    };
+
+    users.groups.gns3 = { };
+
+    services.gns3-server.settings = lib.mkMerge [
+      {
+        Server = {
+          appliances_path = "/var/lib/gns3/appliances";
+          configs_path = "/var/lib/gns3/configs";
+          images_path = "/var/lib/gns3/images";
+          projects_path = "/var/lib/gns3/projects";
+          symbols_path = "/var/lib/gns3/symbols";
+
+          ubridge_path = lib.mkIf (cfg.ubridge.enable) "${cfg.ubridge.package}/bin/ubridge";
+
+          auth = if cfg.auth.enable then "True" else "False";
+          user = lib.mkIf (cfg.auth.user != null) cfg.auth.user;
+          password = lib.mkIf (cfg.auth.passwordFile != null) "@AUTH_PASSWORD@";
+        };
+      }
+      (lib.mkIf (cfg.vpcs.enable) {
+        VPCS.vpcs_path = lib.mkIf (cfg.vpcs.enable) "${cfg.vpcs.package}/bin/vpcs";
+      })
+      (lib.mkIf (cfg.dynamips.enable) {
+        Dynamips.dynamips_path = lib.mkIf (cfg.dynamips.enable) "${cfg.dynamips.package}/bin/dynamips";
+      })
+    ];
+
+    systemd.services.gns3-server = let
+      commandArgs = "--config /etc/gns3/gns3_server.conf"
+        + " --host ${cfg.host}"
+        + " --pid /run/gns3/server.pid"
+        + " --port ${toString cfg.port}"
+        + lib.optionalString (cfg.log.file != null) " --log ${cfg.log.file}"
+        + lib.optionalString (cfg.log.debug) " --debug"
+        + lib.optionalString (cfg.tls.enable) " --ssl"
+        + lib.optionalString (cfg.tls.certFile != null) " --certfile ${cfg.tls.certFile}"
+        + lib.optionalString (cfg.tls.keyFile != null) " --certkey ${cfg.tls.keyFile}"
+      ;
+    in
+    {
+      description = "GNS3 Server";
+
+      after = [ "network.target" "network-online.target" ];
+      wantedBy = [ "multi-user.target" ];
+      wants = [ "network-online.target" ];
+
+      preStart = ''
+        install -m660 ${configFile} /etc/gns3/gns3_server.conf
```

Comment:
We usually use `environment.etc."gns3/gns3_server.conf".source = configFile` for that AFAIK.

Reply:
@RaitoBezarius Hello. Thanks for the review.

Unfortunately I cannot because I use the replace-secret binary on this `/etc/gns3/gns3_server.conf` file, to replace the `@AUTH_PASSWORD@` placeholder by the the password file during pre-start phase of the SystemD service.

Reply:
Then, do you have to make this file global in /etc? Can you make it local to the service rather than global?